lib/rules/curly.js

```diff
@@ -290,7 +290,7 @@ module.exports = {
                 }
             } else if (multiOrNest) {
                 if (hasBlock && body.body.length === 1 && isOneLiner(body.body[0])) {
-                    expected = false;
+                    expected = !!(body.body[0].leadingComments && body.body[0].leadingComments.length);
```

Comment:
I tried making my modification in `isOneLiner` but it caused unit tests to fail. I think the use of isOneLiner in L294 makes modifying the `isOneLiner` function different than my current change.